src/coreWebVitals.ts

```diff
@@ -0,0 +1,160 @@
+	cls: null | number;
+	lcp: null | number;
+	fcp: null | number;
+	ttfb: null | number;
+};
+
+const coreWebVitalsPayload: CoreWebVitalsPayload = {
+	browser_id: null,
+	page_view_id: null,
+	fid: null,
+	cls: null,
+	lcp: null,
+	fcp: null,
+	ttfb: null,
+};
+
+let shouldSendMetrics = false;
+export const bypassSampling = (): void => {
+	shouldSendMetrics = true;
+};
+
+const roundWithDecimals = (value: number, precision = 6): number => {
+	const power = Math.pow(10, precision);
+	return Math.round(value * power) / power;
+};
+
+/**
+ * Send the Core Web Vitals payload to our logging endpoint.
+ *
+ * @param isDev - whether the CODE of PROD endpoint will be used
+ * @returns {boolean} - `true` if the metrics are queued for sending. `false` otherwise
+ */
+const sendData = (isDev: boolean): boolean => {
+	if (!shouldSendMetrics || coreWebVitalsPayload.fcp === null) return false;
+
+	const endpoint = isDev
+		? 'https://performance-events.code.dev-guardianapis.com/core-web-vitals'
+		: 'https://performance-events.guardianapis.com/core-web-vitals';
+
+	return navigator.sendBeacon(endpoint, JSON.stringify(coreWebVitalsPayload));
+};
+
+const onReport: ReportHandler = (metric) => {
+	switch (metric.name) {
+		case 'FCP':
+			// Browser support: Chromium, Firefox, Safari Technology Preview
+			coreWebVitalsPayload.fcp = roundWithDecimals(metric.value);
+			break;
+		case 'CLS':
+			// Browser support: Chromium,
+			coreWebVitalsPayload.cls = roundWithDecimals(metric.value);
+			break;
+		case 'LCP':
+			// Browser support: Chromium
+			coreWebVitalsPayload.lcp = roundWithDecimals(metric.value);
+			break;
+		case 'FID':
+			// Browser support: Chromium, Firefox, Safari, Internet Explorer (with the polyfill)
+			coreWebVitalsPayload.fid = roundWithDecimals(metric.value);
+			break;
+		case 'TTFB':
+			// Browser support: Chromium, Firefox, Safari, Internet Explorer
+			coreWebVitalsPayload.ttfb = roundWithDecimals(metric.value);
+			break;
+	}
+};
+
+type InitCoreWebVitalsOptions = {
+	isDev: boolean;
+
+	browserId?: string | null;
+	pageViewId?: string | null;
+
+	sampling?: number;
+	bypassSampling?: boolean;
+};
+
+/**
+ * Initialise sending Core Web Vitals metrics to a logging endpoint.
+ *
+ * @param init - the initialisation options
+ * @param init.isDev - used to determine whether to use CODE or PROD endpoints.
+ * @param init.browserId - identifies the browser. Usually available via `getCookie('bwid')`. Defaults to `null`
+ * @param init.pageViewId - identifies the page view. Usually available on `guardian.ophan.pageViewId`. Defaults to `null`
+ *
+ * @param init.sampling - sampling rate for sending data. Defaults to `0.01`.
+ * @param init.bypassSampling - send data regardless of sampling rate. Defaults to `false`.
+ *
+ * @param metricsSentCallback - Optional callback, triggered after metrics are queued for sending
+ */
+export const initCoreWebVitals = (
+	{
+		browserId = null,
+		pageViewId = null,
+		bypassSampling = false,
```

Comment:
this can be handled by setting `{sampling: 1}` - is it needed? 

(in theory you could set `{sampling:0.1, bypassSampling:true}` which feels weird)

Reply:
In case my comment above is missed we absolutely need data that ties it to a page view. Otherwise it becomes worthless in some cases.